### DISCUSSION TITLE
Why not just change owners on osiset/laravel-shopify?

### DISCUSSION THREAD

**Discussion by frankiejarrett** [Kyon147/laravel-shopify#20] [General]
Really love the project and glad to see it still has a maintainer and looking forward to becoming a sponsor.

With so many existing devs using this package I was just curious why the original repository didn't just change ownership rather than having to continue on a fork?

Apologies if this has already been discussed elsewhere, I couldn't find anything.

**Comment by Kyon147** [comment]
Hi @fjarrett 

That is a good question.

Tyler and I discussed it and moving over to a fork seemed the cleanest way at the time. I've since detached the fork to be standalone so that there is a more distinct definition between the previous one and this one.

I've kept the namespace as `Osiset` for now, so anyone who wishes to continue to get updates can basically swap their composer item to be `kyon147/laravel-shopify` and then continue as normal as the version numbers I have kept incremental from the previous package. 

Personally I believe leaving the original package as archived is the right move.  Tyler put a lot of his own personal time into the package and keeping archiving it allows him to take a snapshot of his hard work.

**Comment by frankiejarrett** [comment]
@Kyon147 Thanks for the deets! Technically speaking, your approach definitely works.

I guess since Tyler made a succession plan for you to take it over and wasn't just abandoning it, practically speaking, hard forking actually creates more problems than it solves for the community of people using it. Reconfigure composer for updates, issues, discussions, PR history, GitHub stars - all of that lost. Hyperlinks around the net (Stack Overflow, etc.) still pointing to the old repo. Even this fork still has links pointing to the old Wiki docs.

Again, just my perspective, but I think the extra overhead created by forking far outweighs any benefit. I thought maybe there was a special reason for why since it was out of the ordinary when projects change maintainers. It seems like if the community were to vote on it they would take the path of least resistance, but you are the new maintainer so it's your call.

**Reply by Kyon147** [reply]
Totally understand your view, in hindsight I probably would have had the repo owner changed but as they say hindsight is 20/20.

The main reason originally was because I use this package personally, so the fork was just a personal one to keep my own apps open - then I decided to keep the package going for the wider community. 

Most of the PRs got merged before the move, baring maybe one or two - so understand there is some duplication of work now needed but composer is luckily an easy change and the wiki I have now migrated over to this repo rather than linking off.

Hopefully, the slight bit of resistance of changing a couple things on the project will be outweighs by most for the continued maintenance of the package overall 🤞
